### PR TITLE
Fix xielu sharding error

### PR DESCRIPTION
**Pull request by TJ-Solergibert**
This PR removes all the sharding information from the xielu & co. parameters. As they aren't sharded among the TP axis, we just replicate them among TP & DP. Tested directly with the 70B model, we can store and load checkpoints properly.

I will push some little docs regarding how to properly add new modules to Megatron, but in short: 
- If your new module doesn't require any sharding --> If you go with torch.nn.Module as the [`RotaryEmbedding`](https://github.com/swiss-ai/Megatron-LM/blob/9d248e8b5766c8b24649b2b02a9a4ac0b0a3eb5e/megatron/core/models/common/embeddings/rotary_pos_embedding.py#L35) Class it will work. This module belongs to a MegatronModule (GPTModel --> LaunguageModel) which will either call `state_dict` or `sharded_state_dict` when necessary (The module is sharded or not)
  - Why can't Xielu be a torch.nn.Module then? Because the MLP MegatronModule block which contains the Xielu block is [overriding the `sharded_state_dict` method to directly call `sharded_state_dict` in all the blocks belonging to MLP](https://github.com/swiss-ai/Megatron-LM/blob/9d248e8b5766c8b24649b2b02a9a4ac0b0a3eb5e/megatron/core/transformer/mlp.py#L145). We could also create a `sharded_state_dict` within Xielu but IMO it's cleaner this way where `MegatronModule.sharded_state_dict` is calling `state_dict` instead of `sharded_state_dict`.
 - If your new module requires sharding across TP --> Don't try to invent the wheel: Just copy this methods either if you are doing [Column parallelism](https://github.com/swiss-ai/Megatron-LM/blob/9d248e8b5766c8b24649b2b02a9a4ac0b0a3eb5e/megatron/core/extensions/transformer_engine.py#L495-L500) or [Row parallelism ](https://github.com/swiss-ai/Megatron-LM/blob/9d248e8b5766c8b24649b2b02a9a4ac0b0a3eb5e/megatron/core/extensions/transformer_engine.py#L579-L584)
 - Proposed general guideline: Always inherit from `MegatronModule` and just override the `sharded_state_dict` method if the model is sharded across TP.

### CODE REVIEW

**Review by ischlag — APPROVED** [review]
lgtm

**Comment by ischlag** [comment]
The only drawback is that we'll have to convert the checkpoints of the smaller runs before they can be loaded.
